app/src/main/java/org/simple/clinic/medicalhistory/ui/MedicalHistoryDiagnosisWithTreatment.kt

```diff
@@ -0,0 +1,122 @@
+                    bottom = dimensionResource(R.dimen.spacing_4)
+                )
+        ) {
+            Text(
+                modifier = Modifier
+                    .padding(bottom = dimensionResource(R.dimen.spacing_4)),
+                text = diagnosisLabel,
+                style = SimpleTheme.typography.subtitle1Medium,
+                color = MaterialTheme.colors.onSurface,
+            )
+
+            AnswerChipsGroup(
+                modifier = Modifier.align(Alignment.End),
+                selectedAnswer = diagnosisAnswer,
+                onSelectionChange = { newAnswer ->
+                    if (newAnswer == diagnosisAnswer) {
+                        onSelectionChange(diagnosisQuestion, Answer.Unanswered)
+                    } else {
+                        onSelectionChange(diagnosisQuestion, newAnswer)
+                    }
+                }
+            )
+
+            AnimatedVisibility(
+                visible = showTreatmentQuestion
+            ) {
+                Column {
+                    Divider(
+                        modifier = Modifier.padding(
+                            top = dimensionResource(R.dimen.spacing_4),
+                            bottom = dimensionResource(R.dimen.spacing_8)
+                        ),
+                        color = SimpleTheme.colors.onSurface11
+                    )
+
+                    Text(
+                        modifier = Modifier
+                            .padding(bottom = dimensionResource(R.dimen.spacing_4)),
+                        text = stringResource(treatmentQuestion.questionRes),
+                        style = MaterialTheme.typography.body1,
+                        color = MaterialTheme.colors.onSurface,
+                    )
+
+                    AnswerChipsGroup(
+                        modifier = Modifier.align(Alignment.End),
+                        selectedAnswer = treatmentAnswer,
+                        onSelectionChange = { newAnswer ->
+                            if (newAnswer == treatmentAnswer) {
+                                onSelectionChange(treatmentQuestion, Answer.Unanswered)
+                            } else {
+                                onSelectionChange(treatmentQuestion, newAnswer)
+                            }
+                        }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun MedicalHistoryDiagnosisWithTreatmentPreview() {
+    SimpleTheme {
+        MedicalHistoryDiagnosisWithTreatment(
+            diagnosisQuestion = MedicalHistoryQuestion.DiagnosedWithHypertension,
+            diagnosisLabel = stringResource(R.string.medicalhistory_diagnosis_hypertension_required),
+            diagnosisAnswer = Answer.Yes,
+            treatmentQuestion = IsOnHypertensionTreatment(Country.INDIA),
+            treatmentAnswer = Answer.Unanswered,
+            showTreatmentQuestion = true,
+            onSelectionChange = { _, _ -> }
```

Comment:
For my curiosity, can we leave out this option, or is it required? Because it looks like we are doing nothing `onSelectionChange`